src/go/rdctl/pkg/factoryreset/delete_data.go

```diff
@@ -0,0 +1,307 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package factoryreset
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"syscall"
+
+	dockerconfig "github.com/docker/docker/cli/config"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/directories"
+	"github.com/sirupsen/logrus"
+)
+
+func DeleteData(removeKubernetesCache bool) error {
+	return map[string]func(bool) error{
+		"darwin":  deleteDarwinData,
+		"linux":   deleteUnixData,
+		"windows": unregisterAndDeleteWindowsData,
+	}[runtime.GOOS](removeKubernetesCache)
+}
+
+func getStandardDirs() (string, string, string, error) {
+	configDir, err := os.UserConfigDir()
+	if err != nil {
+		return "", "", "", err
+	}
+	cacheDir, err := os.UserCacheDir()
+	if err != nil {
+		return "", "", "", err
+	}
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return "", "", "", err
+	}
+	return configDir, cacheDir, homeDir, nil
+}
+
+func deleteDarwinData(removeKubernetesCache bool) error {
+	configDir, cacheDir, homeDir, err := getStandardDirs()
+	if err != nil {
+		return err
+	}
+	libraryPath := path.Join(homeDir, "Library")
+
+	altAppHomePath := path.Join(homeDir, ".rd")
+	appHomePath := path.Join(configDir, "rancher-desktop")
+	cachePath := path.Join(cacheDir, "rancher-desktop")
+	logsPath := os.Getenv("RD_LOGS_DIR")
+	if logsPath == "" {
+		logsPath = path.Join(libraryPath, "Logs", "rancher-desktop")
+	}
+	settingsPath := path.Join(libraryPath, "Preferences", "rancher-desktop")
+	updaterPath := path.Join(configDir, "Caches", "rancher-desktop-updater")
+
+	pathList := []string{
+		altAppHomePath,
+		appHomePath,
+		logsPath,
+		settingsPath,
+		updaterPath,
+	}
+	if removeKubernetesCache {
+		pathList = append(pathList, cachePath)
+	}
+	return deleteUnixLikeData(homeDir, altAppHomePath, path.Join(homeDir, ".config"), pathList)
+}
+
+func deleteUnixData(removeKubernetesCache bool) error {
```

Comment:
This is actually Linux-specific.

Reply:
ok... I misunderstood what you were getting at. Got it now. Because of course, darwin descends from freebsd, and freebsd + linux are indeed unix-like